source/MaterialXRuntime/RtNameResolver.h

```diff
@@ -0,0 +1,126 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_RTNAMERESOLVER_H
+#define MATERIALX_RTNAMERESOLVER_H
+
+#include <MaterialXRuntime/RtToken.h>
+
+#include <MaterialXCore/Element.h>
+
+#include <memory>
+
+/// @file
+/// TODO: Docs
+
+namespace MaterialX
+{
+
+/// Shared pointer to an RtStringResolverPair
+using RtStringResolverPairPtr = std::shared_ptr<class RtStringResolverPair>;
+
+/// Shared pointer to an RtNameResolverRegistrar
+using RtNameResolverRegistrarPtr = std::shared_ptr<class RtNameResolverRegistrar>;
+
+/// Function for resolving a given token of a given type
+typedef RtToken (*RtNameResolverFunction)(const RtToken& str, const RtToken& type);
+
+/// @class RtNameResolverContext
+///
+/// \brief A structure containing information for custom name resolving
+/// to MaterialX or from MaterialX.
+///
+/// In the structure it is possible to specify a function to call to perform
+/// token changes. Additionally it is possible to supply a set of token
+/// replacements. This is for both to and from MaterialX resolution.
+///
+struct RtNameResolverInfo
+{
+    /// Element type enumeration
+    enum ElementType {
+        FILENAME_TYPE, /// Filename element type
+        GEOMNAME_TYPE  /// Geometry element type
+    };
+
+    RtNameResolverInfo() = default;
+    RtToken identifier {""}; /// Unique token identifier for this information
+    ElementType elementType; /// Type of element this resolver applies to
+    RtNameResolverFunction toFunction; /// Resolver function to resolve to MaterialX. Can be null.
+    RtNameResolverFunction fromFunction; /// Resolver function to resolve from MaterialX. Can be null.
+    RtTokenMap<RtToken> toSubstitutions; /// Custom token substitutions to MaterialX. May be empty.
+    RtTokenMap<RtToken> fromSubstitutions; /// Custom token substitutions from MaterialX. May be empty.
+};
+
+/// @class RtStringResolverPair
+///
+/// Class that keeps track of a pair of String Resolvers that are used to
+/// convert string identifiers to and from MaterialX.
+///
+class RtStringResolverPair
```

Comment:
The string resolver pair that is used to convert string identifiers to/from materialX.

Reply:
If I'm not mistaken this class is never used in the public API. Maybe it could be renamed to PvtStringResolverPair and move into private API?